openedx/frontend-app-gradebook
# feat: enabling header override

**What changed?**

This PR adds `frontend-component-header` as a dependency in the Gradebook MFE and inserts this component to replace the default Gradebook header. It enables the header override for styling/theming purposes. A couple of notes about this change:

- The `frontend-component-header` version used is 2.2.5 to get compatibility with the current Gradebook state. `frontend-component-footer` version is updated to the latest available since it does not have compatibility issues.
- The `AppProvider` component from `frontend-platform` replaces the regular `provider` component in the MFE so the application has the context of the logged-in user and renders the header properly.
-  Some tests were adjusted based on these changes.

**Screenshots**:

![Screenshot from 2021-12-16 09-00-18](https://user-images.githubusercontent.com/15217219/146388891-abddd7d8-0a87-4538-9d4f-8cf2e64c10e4.png)

**Responsive**

![Screenshot from 2021-12-16 09-01-07](https://user-images.githubusercontent.com/15217219/146388908-faa4b430-7d3e-4fc0-87e7-795f6c5728d7.png)

**Developer Checklist**

**Testing Instructions**

There's no particular action to perform when testing this feature. Just run the MFE with this branch and check the header is rendered as expected

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
